### PR TITLE
Gift Card tutorial

### PR DESCRIPTION
**Pull request by rvcas**
This is a tutorial for the gift card example found in the main aiken repo

https://github.com/aiken-lang/aiken/tree/main/examples/gift_card

closes https://github.com/aiken-lang/aiken/issues/461

### CODE REVIEW

File: src/pages/example--gift-card.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  fn gift_card(rdmr, ctx: ScriptContext) -> Bool {
+    todo @"mint and burn"
+  }
+}
+
+validator {
+  fn redeem(d, r, ctx: ScriptContext) -> Bool {
+    todo @"redeem"
+  }
+}
```
Comment (KtorZ) [inline src/pages/example--gift-card.mdx:77]: What about using this example to demonstrate a combined validator?

Comment (rvcas) [reply src/pages/example--gift-card.mdx:77]: @KtorZ I thought about it and I don't want to redo the corresponding example in the main aiken repo.

So I decided we'll just make a new example with the multi-validator and then follow that up with it's own tutorial.

File: src/pages/example--gift-card.mdx [hunk 2]
```diff
@@ -0,0 +1,1120 @@
...
+
+# Gift Card
+
+Let's build a UI to send and redeem a gift card using smart contracts on Cardano.
+
+## Covered in this tutorial
+
+- writing `Aiken`
+- parameterizing scripts
+- using [lucid](https://lucid.spacebudz.io/) (requires a [blockfrost](https://blockfrost.io/) account)
```
Comment (KtorZ) [inline src/pages/example--gift-card.mdx:11]: Just a thought: we could also show a setup using Kupo+Ogmios via Demeter.

Comment (rvcas) [reply src/pages/example--gift-card.mdx:11]: I'm down for that

Comment (rvcas) [reply src/pages/example--gift-card.mdx]: done!

File: src/pages/example--gift-card.mdx [hunk 3]
```diff
@@ -0,0 +1,1137 @@
+import { Callout } from "nextra-theme-docs";
+
+# Gift Card
+
+Let's build a UI to send and redeem a gift card using smart contracts on Cardano.
+
+## Covered in this tutorial
```
Comment (KtorZ) [inline src/pages/example--gift-card.mdx:7]: Nice idea, and something we could generalize for all tutorials.

File: src/pages/example--gift-card.mdx [hunk 4]
```diff
@@ -0,0 +1,1137 @@
...
+This time, for lucid, we'll be using the `Kupmios` provider. You'll need a url
+for [kupo](https://github.com/CardanoSolutions/kupo) and [ogmios](https://github.com/CardanoSolutions/ogmios).
+Thankfully it's very easy to get access to an instance of each on [Demeter](https://demeter.run).
+
+<Callout type="info">
+  ★ Demeter has a generous free tier perfect for this situation. Once you have
+  an account you'll need to create a project. In your project dashboard click
+  `Browse Extensions`, there you'll be able to enable ogmios and kupo. In the
+  screen for each connected instance you can expose a url to access the
+  instance. You'll need these urls later.
```
[the remark this answers is not in the document]
Comment (rvcas) [reply src/pages/example--gift-card.mdx:28]: Ok no problem

File: src/pages/example--gift-card.mdx [hunk 5]
```diff
@@ -0,0 +1,1137 @@
...
+    let ScriptContext { transaction, .. } = ctx
+
+    let Transaction { mint, .. } = transaction
+
+    expect [(asset_name, amount)] =
+      mint
+      |> value.tokens(policy_id)
+      |> dict.to_list()
+
+    amount == -1 && asset_name == token_name
```
[the remark this answers is not in the document]
Comment (rvcas) [reply src/pages/example--gift-card.mdx:344]: Makes sense

**Review by KtorZ — APPROVED** [review]

**Pull request merged**